cmd/ignite/run/exec.go

```diff
@@ -0,0 +1,154 @@
+package run
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"os"
+	"path"
+	"time"
+
+	api "github.com/weaveworks/ignite/pkg/apis/ignite"
+	"github.com/weaveworks/ignite/pkg/constants"
+	"github.com/weaveworks/ignite/pkg/util"
+
+	"golang.org/x/crypto/ssh"
+	shellescape "gopkg.in/alessio/shellescape.v1"
+)
+
+type ExecFlags struct {
+	Timeout      uint32
+	IdentityFile string
+}
+
+type execOptions struct {
+	*ExecFlags
+	vm      *api.VM
+	command []string
+}
+
+func (ef *ExecFlags) NewExecOptions(vmMatch string, command ...string) (eo *execOptions, err error) {
+	eo = &execOptions{
+		ExecFlags: ef,
+		command:   command,
+	}
+	eo.vm, err = getVMForMatch(vmMatch)
```

Comment:
There could be a blank line between the closing brace and `eo.vm` to help readability